src/webmock/stub.cr

```diff
@@ -12,6 +12,10 @@ class WebMock::Stub
     @status = 200
     @body = ""
     @headers = HTTP::Headers{"Content-length" => "0"}
+
+    @block = Proc(HTTP::Request, HTTP::Client::Response).new do |request|
```

Comment:
To reduce complexity under the hood, I decided to always call a block in `#exec`, but just by default have a block which calls out to the `@status`, `@body` and `@headers` instance variables.